sound/soc/sof/sof-audio.c

```diff
@@ -181,12 +189,34 @@ int sof_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 	if (++swidget->use_count > 1)
 		return 0;
 
+	/*
+	 * The scheduler widget for a pipeline is not part of the connected DAPM
+	 * widget list and it needs to be set up before the widgets in the pipeline
+	 * are set up. The use_count for the scheduler widget is incremented for every
+	 * widget in a given pipeline to ensure that it is freed only after the last
+	 * widget in the pipeline is freed. Skip setting up pipeline widget for static pipelines.
+	 */
+	if (swidget->id == snd_soc_dapm_scheduler || !swidget->dynamic_pipeline_widget)
```

Comment:
Can we use the same test as in line 160 to make things a bit more easier to read?
```c
if (swidget->dynamic_pipeline_widget && swidget->id != snd_soc_dapm_scheduler) {
	if (!swidget->pipe_widget) {
		dev_err(sdev->dev, "No scheduler widget set for %s\n", swidget->widget->name);
		ret = -EINVAL;
		goto use_count_dec;
	}

	ret = sof_widget_setup(sdev, swidget->pipe_widget);
	if (ret < 0)
		goto use_count_dec;
}
```